File: algorithms/leetcode/lc003_longest_substring_without_repeating_characters/LC003LongestSubstringWithoutRepeatingCharacters.Tests.cs
using Xunit;

namespace AlgorithmDesign.leetcode.lc003_longest_substring_without_repeating_characters
{
    public class SolutionTests
    {
        [Fact]
        public void TestAbcabcbb()
        {
            Assert.Equal(3, new Solution().LengthOfLongestSubstring("bcabcbb"));
        }

        [Fact]
        public void TestBbbbb()
        {
            Assert.Equal(1, new Solution().LengthOfLongestSubstring("bbbbb"));
        }

        [Fact]
        public void TestDvdf()
        {
            Assert.Equal(3, new Solution().LengthOfLongestSubstring("dvdf"));
        }
    }
}

File: algorithms/code/leetcode/lc003_longest_substring_without_repeating_characters/LC003LongestSubstringWithoutRepeatingCharactersTests.java
package leetcode.lc003_longest_substring_without_repeating_characters;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC003LongestSubstringWithoutRepeatingCharactersTests {
    @Test
    public void testAbcabcbb() throws Exception {
        LC003LongestSubstringWithoutRepeatingCharacters solution;
        solution = new LC003LongestSubstringWithoutRepeatingCharacters();
        assertEquals(3, solution.lengthOfLongestSubstring("abcabcbb"));
    }

    @Test
    public void testBbbbb() throws Exception {
        LC003LongestSubstringWithoutRepeatingCharacters solution;
        solution = new LC003LongestSubstringWithoutRepeatingCharacters();
        assertEquals(1, solution.lengthOfLongestSubstring("bbbbb"));
    }

    @Test
    public void testDvdf() throws Exception {
        LC003LongestSubstringWithoutRepeatingCharacters solution;
        solution = new LC003LongestSubstringWithoutRepeatingCharacters();
        assertEquals(3, solution.lengthOfLongestSubstring("dvdf"));
    }
}
